### PR TITLE
Exploration/cogsci2020a model inventory

### PR DESCRIPTION
**Pull request by hollandjg**
## Description

First version of a model inventory using a more functional approach, building on the awesome cogsci work.

part of #232

### Type of change:
- New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Comment by hollandjg** [comment]
I had a look at the work @musslick has done in the cogsci directory. It's pretty neat. It inspired me to look at how we could use a functional approach to access the models and the plotters in a more future-proof way. 

Might not be useful right now, but I think this is one direction we could go for the long-term model registry / inventory.

File: studies/cogsci2023/models/prospect_theory.py [hunk 1]
```diff
@@ -183,3 +184,11 @@ def plot_prospect_theory(model=None):
 # import matplotlib.pyplot as plt
 # plt.plot(x_range, y)
 # plt.show()
+
+register_model(
+    "prospect_theory",
+    metadata=prospect_theory_metadata,
+    data=prospect_theory_data,
+    synthetic_experiment_runner=prospect_theory_experiment,
+    name="Prospect Theory",
```
Comment (gtdang) [inline studies/cogsci2023/models/prospect_theory.py:193]: Do we need the plotter added here?

File: studies/cogsci2023/models/weber_fechner.py [hunk 2]
```diff
@@ -108,3 +101,12 @@ def plot_weber_fechner(model = None):
...
 
 # X, y = weber_fechner_data(weber_fechner_metadata())
+
+
+register_model(
+    "weber_fechner",
+    metadata=weber_fechner_metadata,
+    data=weber_fechner_data,
+    synthetic_experiment_runner=weber_fechner_experiment,
+    name="Weber-Fechner Law",
```
Comment (gtdang) [inline studies/cogsci2023/models/weber_fechner.py:111]: Do we need the plotter added here?

**Review by gtdang — COMMENTED** [review]
I like the implementation of registering and retrieving models here. Curious would this also work if you just import specific models and not the entire model directory? I think it should if I'm interpreting it correctly.

File: studies/cogsci2023/models/prospect_theory.py (hunk 1 above)
Comment (hollandjg) [reply studies/cogsci2023/models/prospect_theory.py:193]: > Curious would this also work if you just import specific models and not the entire model directory?

Absolutely. You could register single models at a time (and that's probably the best way to refactor it – just add the models one by one), or register your own models if you like, and have them available to anything which uses the model inventory.

I envisage us loading all of the local models into the inventory whenever we load the autora.model submodule, but you could also have a situation where for some pre-registered models which have a lot of data there's a placeholder which then loads the model on the fly from a website, _a la_ scikit-learn.

File: studies/cogsci2023/models/weber_fechner.py (hunk 2 above)
Comment (hollandjg) [reply studies/cogsci2023/models/weber_fechner.py:111]: good catch, yes we do.

File: studies/cogsci2023/models/prospect_theory.py (hunk 1 above)
Comment (hollandjg) [reply studies/cogsci2023/models/prospect_theory.py:193]: > Do we need the plotter added here?

Good catch! Updated.

Comment (gtdang) [reply studies/cogsci2023/models/prospect_theory.py:193]: Great!

**Comment by hollandjg** [comment]
replaced by #250

**Pull request closed without merging**